reactioncommerce/reaction-next-starterkit
# refactor: navigation data comes from new NavigationTree

Resolves #446
Impact: **minor**
Type: **refactor**

## Issue
As a Reaction storefront user, I can view the navigation built by the admin, so that I can see a nice custom Navigation.

## Solution
This is the UI portion of https://github.com/reactioncommerce/reaction/pull/4683
This PR updates the starterkit to use the `NavigationTree` to display navigation, rather than tags.

## Breaking changes
- Any store which uses tags as their navigation will need to build out a new Navigation Tree following the instructions in https://github.com/reactioncommerce/reaction/pull/4683
- Any custom code which was using the `primaryShopID` gql query will need to update this to use the `primaryShop` query, and use the `shop._id` from that data

## Testing
1. Follow the instructions in https://github.com/reactioncommerce/reaction/pull/4683 to create a NavigationTree
1. See that the UI displays this new tree as the navigation, instead of using existing tags
1. Change the language to `es` (or another language, if you've added translations) [here](https://github.com/reactioncommerce/reaction-next-starterkit/blob/refactor-kieckhafer-446-navigation/src/lib/data/getNavigationTree.js#L26), and see that the nav updates for items that have translations
